types/pouchdb-core/index.d.ts

```diff
@@ -507,45 +503,33 @@ declare namespace PouchDB {
          * which should match the ID and revision of the document on which to base your updates.
          * Finally, to delete a document, include a _deleted parameter with the value true.
          */
-        bulkDocs(docs: Core.PutDocument<Content>[],
-                 options?: Core.BulkDocsOptions): Promise<Core.Response[]>;
+        bulkDocs<Model>(docs: Array<Core.PutDocument<Content & Model>>,
+                        options?: Core.BulkDocsOptions): Promise<Core.Response[]>;
 
         /** Compact the database */
         compact(options?: Core.CompactOptions): Promise<Core.Response>;
 
         /** Compact the database */
         compact(options: Core.CompactOptions,
-                callback: Core.Callback<Core.Error, Core.Response>): void;
+                callback: Core.Callback<Core.Response>): void;
 
         /** Destroy the database */
-        destroy(options: Core.DestroyOptions | void,
-            callback: Core.AnyCallback): void;
+        destroy(options: Core.Options | null,
+                callback: Core.Callback<any>): void;
 
         /** Destroy the database */
-        destroy(options?: Core.DestroyOptions | void): Promise<void>;
+        destroy(options?: Core.Options | null): Promise<void>;
 
         /** Fetch a document */
-        get(docId: Core.DocumentId,
-            options: Core.GetOptions | null,
-            callback: Core.Callback<any, Core.Document<Content> & Core.GetMeta>
-            ): void;
+        get<Model>(docId: Core.DocumentId,
+                   options: Core.GetOptions | Core.GetOpenRevisions | null,
+                   callback: Core.Callback<Core.Document<Content & Model> & Core.GetMeta>
+                  ): void;
 
         /** Fetch a document */
-        get(docId: Core.DocumentId,
-            options: Core.GetOptions
-            ): Promise<Core.Document<Content> & Core.GetMeta>;
-
-        /** Fetch a document */
-        get(docId: Core.DocumentId): Promise<Core.Document<Content> & Core.GetMeta>;
-
-        /** Fetch document open revs */
-        get(docId: Core.DocumentId,
-            options: Core.GetOpenRevisions,
-            callback: Core.Callback<any, Core.Revision<Content>[]>): void;
-
-        /** Fetch document open revs */
-        get(docId: Core.DocumentId,
-            options: Core.GetOpenRevisions): Promise<Core.Revision<Content>[]>;
+        get<Model>(docId: Core.DocumentId,
+                   options?: Core.GetOptions | Core.GetOpenRevisions
+                  ): Promise<Core.Document<Content & Model> & Core.GetMeta>;
```

Comment:
Yes, I was too aggressive and I missed the `Core.Revision` difference in interpreting the linter's signature merges.